### PR TITLE
blog: deploying OpenShift Origin Stand-alone Registry

### PR DESCRIPTION
**Pull request by miabbott**
This blog post details how to use the OpenShift Origin installer to
deploy a stand-alone registry on Fedora 25 Atomic Host.  This can be
used as a fallback source of documentation in case the Atomic Registry
installer does not work successfully.

### CODE REVIEW

**Comment by miabbott** [comment]
cc: @aweiteka @dustymabe

**Comment by dustymabe** [comment]
so this is an alternative to the docs that already exist for the atomic registry? almost seems like we should fix those docs instead?

**Comment by jasonbrooks** [comment]
The atomic registry is broken, and when people ask in #atomic -- basically every single day -- they're told to use the standalone registry instead. I think it's atomic registry that needs fixing or straight up retirement.

**Comment by dustymabe** [comment]
> I think it's atomic registry that needs fixing or straight up retirement.

i agree with that

**Comment by jberkus** [comment]
@aweiteka, wanna weigh in on this?

**Comment by miabbott** [comment]
> so this is an alternative to the docs that already exist for the atomic registry? almost seems like we should fix those docs instead?

I briefly discussed this in person with @aweiteka and my impression was that he didn't have the bandwidth necessary to maintain the current Atomic Registry installer and/or always be available for the questions that come up on #atomic.

After spending the day getting familiar with the stand-alone registry installation process via OpenShift Origin, I think moving to the OO Registry instead of the Atomic Registry wouldn't be a difficult change.  
This blog post basically gets you to a similar POC instance that you would get from using the current Atomic Registry docs.

**Comment by dustymabe** [comment]
can we please schedule some time to chat about this? with all the people on this ticket? we also probably need to start a weekly projectatomic meeting where the projects can sync.

**Comment by miabbott** [comment]
> can we please schedule some time to chat about this? with all the people on this ticket? we also probably need to start a weekly projectatomic meeting where the projects can sync.

Can we add it to the atomic-wg meeting agenda?  I know that it is mostly Fedora specific there, but a majority of the involved folks have shown up to that meeting before.

File: source/blog/2017-05-12-oo-standalone-registry.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+Prepping the Installer
+=======================
+OpenShift Origin provides an excellent [Ansible-based](https://www.ansible.com/) installer that we will used to deploy the registry.  So the first thing to do is checkout the installer from the [GitHub repo](https://github.com/openshift/openshift-ansible).  At the time of this writing, I used a specific version ([3.4.24-1](https://github.com/openshift/openshift-ansible/releases/tag/openshift-ansible-3.4.24-1)) of the repo to ensure it would work correctly for me.  We'll checkout the repo to that version as part of this process.
+
+    $ git clone https://github.com/openshift/openshift-ansible.git
+    $ cd openshift-ansible
+    $ git checkout openshift-ansible-3.4.24-1
+    HEAD is now at 90f6a70... Automatic commit of package [openshift-ansible] release [3.6.24-1].
+
+The next step is to prepare the inventory file for the installer.  My dedicated system for the registry has a 'public' IP address of `10.8.172.199`, so we use that value throughout.  (I'm using a VM in a private OpenStack instance, so the 'public' IP address is only visible on our corporate network and is not reachable from the Internet.)
```
Comment (aweiteka) [inline source/blog/2017-05-12-oo-standalone-registry.md]: Could you mention the use of [xip.io](http://xip.io/) here? It's an essential bit of magic to make dns routing work. How about...

"Notice I'm using the [xip.io](http://xip.io/) service to resolve wildcard DNS queries."

**Comment by miabbott** [comment]
From a meeting between @jberkus, @dustymabe, @aweiteka and myself, it was decided we would gather feedback about sunsetting the Atomic Registry and moving towards suggesting the use OpenShift Origin stand-alone registry.  Until the feedback has been gathered, this post is on hold.

File: source/blog/2017-05-12-oo-standalone-registry.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    $ ansible-playbook -i myinventory -e 'ansible_python_interpreter=/usr/bin/python3'
+    ...
+    ...
+    PLAY RECAP *********************************************************************
+    10.8.172.199               : ok=567  changed=130  unreachable=0    failed=0
+    localhost                  : ok=9    changed=0    unreachable=0    failed=0
+
+Running the installer will take a few minutes, but once it finished, we can get right to the registry console.
+
+Assuming you have used a similar value for `openshift_master_default_subdomain`, you'll be able to access your registry console at a URL similar to `https://registry-console-default.10.8.172.199.xip.io/`.  Just append `registry-console-default.` to your `openshift_master_default_subdomain`.
```
Comment (dmsimard) [inline source/blog/2017-05-12-oo-standalone-registry.md]: Note: the registry console host will be made configurable once I clean up and land https://github.com/openshift/openshift-ansible/pull/4181

**Comment by jberkus** [comment]
So it looks like it'll take a while to resolve the Atomic Registry question.  Given that, I'd like to publish the blog post but wait on replacing the Atomic Registry docs.

**Comment by aweiteka** [comment]
> publish the blog post but wait on replacing the Atomic Registry docs.

Sounds like a good approach.

**Comment by dustymabe** [comment]
> So it looks like it'll take a while to resolve the Atomic Registry question.

I'd rather not drag our feet on that considering it is broken and we're just going to get more people that waste their time. Not a good first user experience.

**Comment by dmsimard** [comment]
Feel free to go ahead with this format, I'll work on a more "production ready"-type post later this week, taking into account the experience we've had standing up the registry for RDO. I'll link to this post so people have an easy "proof of concept" route to take.

**Comment by miabbott** [comment]
I've got a few edits to make before publishing this, but I'm a bit swamped in RHEL testing at the moment.

But I still think I can get the done in the next day or two.

**Comment by miabbott** [comment]
Pushed some changes to address feedback from @aweiteka and opened the door for another blog post about running the registry in production.

**Comment by dmsimard** [comment]
We probably want another date than May 12?

**Comment by jberkus** [comment]
I fix the date when I publish.  This will go out today.

**Pull request merged**